packages/cli/lib/model-discoverer.js

```diff
@@ -0,0 +1,105 @@
+const debug = require('./debug')('model-discoverer');
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Given a datasource and discovery options,
+ * return a list of objects {table: 'foo', schema: 'bar}
+ */
+async function discoverModelNames(ds, options) {
+  if (!ds.connected) {
+    await new Promise(resolve => {
+      ds.on('connected', resolve);
+    });
+  }
+  return await ds.discoverModelDefinitions(options);
+}
+
+/**
+ * Returns the entire model definition for a model
+ */
+async function discoverSingleModel(ds, options, name) {
+  const schema = await ds.discoverSchema(name, options);
+  schema.settings = schema.options;
+  return schema;
+}
+
+/**
+ * Loads a DataSource from a file
+ * If the path provided is a JSON, it instantiates a juggler.DataSource with the config as the only argument
+ * Else it requires it like a compiled loopback datasource
+ * @param path
+ * @returns juggler.DataSource
+ */
+function loadDataSource(path) {
+  const ds = require(path);
+  const key = Object.keys(ds)[0];
+  const val = new ds[key]();
+  return val;
+}
+
+/**
+ * Loads a compiled loopback datasource by name
+ * @param name {string}
+ * @returns {*}
+ */
+function loadDataSourceByName(name) {
+  debug(`Searching for specified dataSource ${name}`);
+  const dataSourceFiles = getAllDataSourceFiles();
+  debug(`Loaded ${dataSourceFiles.length} dataSource files`);
+
+  for (let i = 0; i < dataSourceFiles.length; i++) {
+    const f = dataSourceFiles[i];
+    const ds = loadDataSource(path.resolve(DEFAULT_DATASOURCE_DIRECTORY, f));
+    if (ds.name === name) {
+      debug(`Found dataSource ${name}`);
+      return ds;
+    } else {
+      debug(`Did not match dataSource ${name} !== ${ds.name}`);
+    }
+  }
+  throw new Error(
+    `Cannot find datasource "${name}" in ${DEFAULT_DATASOURCE_DIRECTORY}`,
+  );
+}
+
+const DEFAULT_DATASOURCE_DIRECTORY = './dist/src/datasources';
```

Comment:
```suggestion
const DEFAULT_DATASOURCE_DIRECTORY = './dist/datasources';
```

I thought we already have code to discover datasources elsewhere in the CLI codebase, isn't such functionality required by `lb4 repository` command? Can we find a way how to extract a shared helper and avoid code duplication?

Reply:
done